### PR TITLE
adds first pass at an initial glossary

### PR DESCRIPTION
**Pull request by Frijol**
Excited to try the gitbooks glossary feature on our new docs @HipsterBrown what do you think of this as a first pass?

(And: any edits welcome)

### CODE REVIEW

File: GLOSSARY.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+## access point
+Networking hardware which creates a local area network over Wifi
```
Comment (HipsterBrown) [inline GLOSSARY.md]: This is a somewhat hard feature to describe. Maybe something like:

> Enable Tessel to create/emit a local Wifi network.

?

Comment (HipsterBrown) [inline GLOSSARY.md]: Oh nevermind. I misconstrued the point of this document. 😞 

This is a good definition of an access point. :+1:

File: GLOSSARY.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+## analog
+In an electrical signal, existing in a continuous range, as opposed to the discrete ranges of digital
```
Comment (HipsterBrown) [inline GLOSSARY.md]: What type of components use analog signals?

File: GLOSSARY.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+## API
+Application program interface: a set of tools and protocols which can be used to build software. In the case of Tessel, the hardware API is a library you can use to connect to modules, manipulate LEDs, and call out specific pins
+
+## CLI
+Command line interface: a tool which lets you interact with a program (or in the case of Tessel, with the Tessel) via text-based commands in the command line, or terminal
+
+## digital
+In an electrical signal, existing at discrete values: on or off, zero or one
```
Comment (HipsterBrown) [inline GLOSSARY.md]: discrete/binary?

File: GLOSSARY.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+## GPIO
+General purpose input and output: electrical pins which can be configured for various purposes
+
+## I2C
+A hardware communication protocol which uses two wires: SCL and SDA. Read more about it in the Communication Protocols doc
```
Comment (HipsterBrown) [inline GLOSSARY.md]: Can we link to that documentation?

File: GLOSSARY.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+A standard interface for connecting wires and modules to Tessel, configured as 10 "pins" or points of electrical connection
+
+## pins
+Points of electrical connection with an integrated circuit
+
+## ports
+Interfaces for electrical connections
+
+## PWM
+Pulse-width modulation: a hardware communication protocol where there is a digital signal fluctuating between values at a set period, and the message is encoded by changing the percentage of that period in which the signal stays at one of the values. This can be used to approximate an analog signal on a pin which only supports digital signaling
```
Comment (HipsterBrown) [inline GLOSSARY.md]: What kind of components use this protocol?

File: GLOSSARY.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+## RX
+Receive: the line in a UART interface which receives data from the controlled device
```
Comment (HipsterBrown) [inline GLOSSARY.md]: "wire" was used in earlier definitions, should we standardize on "wire" over "line"?

Comment (HipsterBrown) [inline GLOSSARY.md]: Can we link to the UART definition in this document?

File: GLOSSARY.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+Serial data line: the data-carrying wire of an I2C interface
+
+## serial
+See: UART
+
+## slave
+A device or part of a device which is directly controlled by another
+
+## SPI
+A hardware communication protocol which uses SCK, MISO, MOSI, and SS/CS (chip select) lines. Learn more in the Communication Protocols doc
```
Comment (HipsterBrown) [inline GLOSSARY.md]: Can we link to that documentation?

**Comment by HipsterBrown** [comment]
This is awesome and definitely needed! Thanks for working on it because it will be incredibly useful. 🎉

File: GLOSSARY.md (hunk 4 above)
Comment (Frijol) [inline GLOSSARY.md]: Running into https://github.com/GitbookIO/gitbook/issues/729

File: GLOSSARY.md (hunk 6 above)
Comment (Frijol) [inline GLOSSARY.md]: Unfortunately not, I wish the glossary would cross-link itself.

**Comment by Frijol** [comment]
thanks @HipsterBrown! I responded to your comments.

I've now added markdown links to a few things, too, but due to https://github.com/GitbookIO/gitbook/issues/729 they won't render. What do you think, should I leave them in (since by design markdown is pretty clear  even unrendered) or find a different way to call out the links in plaintext?

**Comment by HipsterBrown** [comment]
That's an odd bug. I think we should convert the links to `<a href=""></a>` tags until that bug is resolved.

**Comment by Frijol** [comment]
ah, good point

**Comment by Frijol** [comment]
@HipsterBrown good to go?

**Comment by HipsterBrown** [comment]
LGTM 👍

**Pull request merged**